tools/bin/hawqregister

```diff
@@ -0,0 +1,272 @@
+        dburl = dbconn.DbURL(hostname=options.host, port=options.port, username=options.user, dbname=databasename)
+        conn = dbconn.connect(dburl, False)
+        rows = dbconn.execSQL(conn, query)
+	conn.commit()
+        conn.close()
+
+    except DatabaseError, ex:
+        logger.error("Failed to connect to database, this script can only be run when the database is up")
+        logger.error("host = %s, port = %d, user = %s, dbname = %s, query = %s" % (options.host, options.port, options.user, databasename, query))
+        sys.exit(1)
+
+
+    # move file(s) in src path into the folder correspoding to the target table
+    if (normal == True):
+        for row in rows:
+            destdir = row[0].strip() + "/" + str(row[1]) + "/" + str(row[2]) + "/" + str(row[3]) + "/"
+
+        segno = firstsegno
+        for file in files:
+            srcfile = file
+            dstfile = destdir + str(segno)
+            segno += 1
+            if srcfile != dstfile:
+                hdfscmd = "hadoop fs -mv %s %s" % (srcfile, dstfile)
+                sys.stdout.write("hafscmd: '%s'\n" % hdfscmd)
+                result = local_ssh(hdfscmd)
+                if result != 0:
+                    logger.error("Fail to move '%s' to '%s'" % (srcfile, dstfile))
+                    sys.exit(1)
+    else:
+        for row in rows:
+            srcdir = row[0].strip() + "/" + str(row[1]) + "/" + str(row[2]) + "/" + str(row[3]) + "/"
+
+        segno = firstsegno
+        for file in files:
+            dstfile = file
+            srcfile = srcdir + str(segno)
+            segno += 1
+            if srcfile != dstfile:
+                hdfscmd = "hadoop fs -mv %s %s" % (srcfile, dstfile)
+                sys.stdout.write("hafscmd: '%s'\n" % hdfscmd)
+                result = local_ssh(hdfscmd)
+                if result != 0:
+                    logger.error("Fail to move '%s' to '%s'" % (srcfile, dstfile))
+                    sys.exit(1)
+
+
+def insert_metadata_into_database(options, databasename, tablename, seg_name, firstsegno, eofs):
+    try:
+        query = "SET allow_system_table_mods='dml';"
+        segno = firstsegno
+        for eof in eofs:
+            query += "insert into pg_aoseg.%s values(%d, %d, %d, %d);" % (seg_name, segno, eof, -1, -1)
+            segno += 1
+
+        dburl = dbconn.DbURL(hostname=options.host, port=options.port, username=options.user, dbname=databasename)
+        conn = dbconn.connect(dburl, True)
+        rows = dbconn.execSQL(conn, query)
+	conn.commit()
+        conn.close()
+
+    except DatabaseError, ex:
```

Comment:
> The main hawq register steps are as follows:
> get_files_in_hdfs(filepath)
> check_parquet_format(options, files)
> firstsegno = get_metadata_from_database(options, databasename, seg_name)
> move_files_in_hdfs(options, databasename, tablename, files, firstsegno, True)
> insert_metadata_into_database(options, databasename, tablename, seg_name, firstsegno, sizes)

If one of the reads fails, we just exit cause no file has been changed. If move_files_in_hdfs fails, the metadata will not be inserted. If insert_metadata_into_database fails, we will move the files back to their original location.